### PR TITLE
WLTS-8&9 - Create a new database tables for Dictionary & TranslationUpload

### PR DESCRIPTION
**Pull request by sullivangareth**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/WLTS-8
https://tools.hmcts.net/jira/browse/WLTS-9

### Change description ###

English phrase to equivalent Welsh phrase translation will be need to be stored in dictionary as part of Welsh language support. To support this a new database table dictionary will be required.

When a new translation CVS is uploaded, this needs to be stored. To support this a new table translation_upload needs be created.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/translate/repository/DictionaryRepositoryIT.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            () -> assertEquals(now, dictionary.getTranslationUpload().getUploaded()),
+            () -> assertEquals(IDAM_USER_ID, dictionary.getTranslationUpload().getUserId())
+        );
+    }
+
+    @Sql(scripts = {DELETE_TRANSLATION_TABLES_SCRIPT, GET_TRANSLATION_TABLES_SCRIPT})
+    @Test
+    void testFindDictionaryAndTranslationUploadNoTranslationPhrase() {
+        final var optionalDictionaryEntity = dictionaryRepository.findById(1);
+        assertTrue(optionalDictionaryEntity.isPresent());
```
Comment (Thor-tech-of-metal) [inline src/integrationTest/java/uk/gov/hmcts/reform/translate/repository/DictionaryRepositoryIT.java:66]: you could use val, instead of final var

**Review by Thor-tech-of-metal — APPROVED** [review]

File: build.gradle [hunk 2]
```diff
@@ -25,7 +25,15 @@ java {
...
 def versions = [
+        junit           : '5.7.2',
+        junitPlatform   : '1.7.2',
+        jsr310          : '2.13.2',
+        lombok          : '1.18.24',
+        reformLogging   : '5.1.7',
         springBoot      : springBoot.class.package.implementationVersion,
+        springCloud     : '2021.0.2',
+        springfoxSwagger: '3.0.0',
+        serviceAuthVersion: '4.0.3',
```
Comment (danlysiak) [inline build.gradle:36]: I think we can delete all these apart from `lombok` that are now showing as new changes under this PR? (Presumably added back in as part of a merge conflict!)

**Review by danlysiak — COMMENTED** [review]
Just a couple of minor comments, but the main purpose of this PR looks good!

File: build.gradle (hunk 2 above)
Comment (sullivangareth) [reply build.gradle]: Yes, merge conflict - fixed in https://github.com/hmcts/ts-translation-service/pull/5/commits/6acfcae56a6cab21d7228b8bccc44c3b51ff2a75

File: src/main/resources/db/migration/V20220503_7__WLST-7-CreateDictionaryAndTranslationUploadDbs.sql [hunk 3]
```diff
@@ -0,0 +1,43 @@
+-- translation_upload table and sequence creation
```
[the remark this answers is not in the document]
Comment (sullivangareth) [reply src/main/resources/db/migration/V20220503_7__WLST-7-CreateDictionaryAndTranslationUploadDbs.sql]: Fixed in https://github.com/hmcts/ts-translation-service/pull/5/commits/6acfcae56a6cab21d7228b8bccc44c3b51ff2a75

File: build.gradle [hunk 4]
```diff
@@ -165,9 +172,14 @@ dependencies {
...
   }
   testImplementation group: 'com.github.hmcts', name: 'befta-fw', version: '8.7.1'
-  integrationTestImplementation group: 'org.testcontainers', name: 'postgresql', version: versions.testcontainers
 
-  runtimeOnly 'org.postgresql:postgresql'
+  implementation group: 'org.flywaydb', name: 'flyway-core', version: '7.7.0'
+  implementation group: 'org.postgresql', name: 'postgresql', version: '42.3.4'
+
+  annotationProcessor group: 'org.projectlombok', name: 'lombok', version: versions.lombok
+  implementation group: 'org.projectlombok', name: 'lombok', version: versions.lombok
```
[the remark this answers is not in the document]
Comment (sullivangareth) [reply build.gradle:173]: fixed in https://github.com/hmcts/ts-translation-service/pull/5/commits/6acfcae56a6cab21d7228b8bccc44c3b51ff2a75

File: src/integrationTest/java/uk/gov/hmcts/reform/translate/repository/DictionaryRepositoryIT.java (hunk 1 above)
Comment (sullivangareth) [reply src/integrationTest/java/uk/gov/hmcts/reform/translate/repository/DictionaryRepositoryIT.java:66]: 'val' is a lombok thing - while I like lombok for annotating classes, I'm not so keen on using it here...will leave as is for  now

**Review by danlysiak — APPROVED** [review]

**Pull request merged**